packages/components/src/decorative-boxes/index.tsx

```diff
@@ -0,0 +1,201 @@
+import style from './style.module.css';
+
+interface BoxPosition {
+  top?: number;
+  left?: number;
+  bottom?: number;
+  right?: number;
+}
+
+export interface DecorativeBoxesProps
+  extends React.ComponentPropsWithoutRef<any> {
+  children: JSX.Element;
+  boxSize?: number;
+  box1Position?: BoxPosition;
```

Comment:
Her ville jeg ha abstrahert det litt for å ikke lekke "detaljer" ut til konsument på et vis f.eks:

```
type BoxPosition = 'top-left' | 'top-right';
```

Eller noe lignende da.

Samme gjelder litt farge også. Hvordan er det om man definerer farge til box1 men ikke posisjon? Eller posisjon men ikke farge? Kanskje er det ok å bare bruke defaults, men signatur-messig så kommer det ikke tydelig frem at disse to har t forhold til hverandre. Det kan f.eks løses med types som noe i retning av:

```ts
type BoxProperties<T extends 'top' | 'bottom'> = {
  position: `${T}-${'left' | 'right'}`;
  color?: string; // or even better: specific color type from the theme
}

type DecorativeBoxesProps = PropsWithChildren<{
 boxSize: 'small' | 'medium' | 'large';
 topBox?: BoxProperties<'top'>;
 bottomBox?: BoxProperties<'bottom'>
}>;
```

Ok, kanskje ikke så avansert da. Jeg ble bare gira på å bruke [template literal types](https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html) 😅 

Men å gruppere dem sammen da i praksis, visa via signaturen at disse egenskapene hører sammen.

Reply:
@mikaelbr Tanker om noe sånt som en løsning? Langt enklere enn å skrive inn piksler/prosenter, men litt mer granulær kontroll enn top/bottom-right/left. Navnene på topish, bottomish, leftish og rightish er også up for discussion.

(Du får unnskylde den rare formateringa, er sånn som dette Prettier vil ha det)
<img width="579" alt="image" src="https://user-images.githubusercontent.com/31799931/186894733-8219f485-dcc2-4fe3-9a64-b8a9dac186b6.png">